rasterio/rio/shapes.py

```diff
@@ -136,6 +139,8 @@ def __call__(self):
                         src.width % sampling, src.height % sampling)
                     # And follow by scaling.
                     transform *= Affine.scale(float(sampling))
+                    shape = (int(math.ceil(src.height / sampling)),
+                             int(math.ceil(src.width / sampling)))
```

Comment:
This results in different x- and y-scaling factors, so the resulting transform isn't correct. Would it make more sense to re-calculate the sampling factor that can actually be used according to the minimum dimension?